README.md

```diff
@@ -237,6 +237,14 @@ without the agent. In order to do that, the `provisionVMAgent` flag must be set
 provisioning time, via whichever API is being used. We will provide more details on
 this on our wiki when it is generally available. 
 
+#### __Provisioning.Force__
```

Comment:
I think it's a bit confusing that waagent can be forced to provision but at the same time it's not enabled. In which case provisioning through cloud-init and waagent will be skipped.
One possible solution is to make the "Force=true" option overrides the "Enabled" option for waagent.

Reply:
The more I think about this, the more I think that we are trying to force a bool where it doesn't belong. I think I'm more inclined to completely remove `Provisioning.Force` and deprecate `Provisioning.Enabled` in favor of a string type `Provisioning.Agent`, defaulting to `auto` but allowing you to specify `waagent` or `cloud-init`.

`auto` behavior could be auto detect, but otherwise it gives the user total control of which provisioning agent they could use which would act as an override. I think this is much more intuitive then trying to combine a few bools to get the desired effect.

What do you think?

Reply:
I like the idea of "Provisioning.Agent". It's more intuitive and easier to grasp. Perhaps it can be added to the conf file with a comment of all the possible values.